app/domain/repos/conjur_ca.rb

```diff
@@ -14,12 +14,26 @@ def self.create(resource_id)
     #
     def self.ca(resource_id)
       ca_info = ::Conjur::CaInfo.new(resource_id)
-      stored_cert = Resource[ca_info.cert_id].last_secret.value
-      stored_key = Resource[ca_info.key_id].last_secret.value
+      stored_cert = fetch_ca_resource(ca_info.cert_id)
+      stored_key = fetch_ca_resource(ca_info.key_id)
       ca_cert = OpenSSL::X509::Certificate.new(stored_cert)
       ca_key = OpenSSL::PKey::RSA.new(stored_key)
       ::Util::OpenSsl::CA.new(ca_cert, ca_key)
     end
 
+    def self.fetch_required_secrets
+      @fetch_required_secrets ||= ::Conjur::FetchRequiredSecrets.new
+    end
+
+    def self.fetch_ca_resource(ca_resource_id)
+      begin
+        fetch_required_secrets.(resource_ids: [ca_resource_id])[ca_resource_id]
+      rescue Errors::Conjur::RequiredResourceMissing
+        raise Errors::Authentication::AuthnK8s::CaResourceMissing, ca_resource_id
+      rescue Errors::Conjur::RequiredSecretMissing
+        raise Errors::Authentication::AuthnK8s::CaResourceValueMissing, ca_resource_id
```

Comment:
do you see value in raising the error `CaResourceValueMissing` instead of the error `RequiredResourceMissing`? what value does the user get from this message that they don't get from the original?